#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_check_snmp_03_state_based.dpatch by
## Thomas Guyot-Sionnest <dermoth@aei.ca>
##
## From ffadb45f924aa2ba9792963bdc64d798841b9e7b Mon Sep 17 00:00:00 2001
## From: Thomas Guyot-Sionnest <dermoth@aei.ca>
## Date: Tue, 30 Nov 2010 22:28:19 -0500
## Subject: [PATCH 3/4] State-based tests enhancements
## (Debian Bug #607736)
##
## DP: State-based tests enhancements

@DPATCH@

diff --git a/plugins/tests/check_snmp.t b/plugins/tests/check_snmp.t
index 08348d2..6966838 100755
--- a/plugins/tests/check_snmp.t
+++ b/plugins/tests/check_snmp.t
@@ -51,6 +51,9 @@ if ($ARGV[0] && $ARGV[0] eq "-d") {
 	}
 }
 
+# We should merge that with $ENV{'NPTEST_CACHE'}, use one dir for all test data
+$ENV{'NAGIOS_PLUGIN_STATE_DIRECTORY'} ||= "/var/tmp";
+
 my $tests = 41;
 if (-x "./check_snmp") {
 	plan tests => $tests;
@@ -106,7 +109,7 @@ like($res->output, '/'.quotemeta('SNMP OK - And now have fun with with this: \"C
 "And now have fun with with this: \"C:\\\\\"
 because we\'re not done yet!"').'/m', "Attempt to confuse parser No.3");
 
-system("rm /usr/local/nagios/var/check_snmp/*");
+system("rm -f ".$ENV{'NAGIOS_PLUGIN_STATE_DIRECTORY'}."/check_snmp/*");
 $res = NPTest->testCmd( "./check_snmp -H 127.0.0.1 -C public -p $port_snmp -o .1.3.6.1.4.1.8072.3.2.67.10 --rate -w 600" );
 is($res->return_code, 0, "Returns OK");
 is($res->output, "No previous data to calculate rate - assume okay");
-- 
1.7.3.2

